crates/ruff_linter/src/rules/pylint/rules/boolean_chained_comparison.rs

```diff
@@ -83,20 +91,44 @@ pub(crate) fn boolean_chained_comparison(checker: &mut Checker, expr_bool_op: &E
                 return None;
             }
 
-            let edit = Edit::range_replacement(
-                left_compare_right.id().to_string(),
-                TextRange::new(left_compare_right.start(), right_compare_left.end()),
+            let left_has_paren = parenthesized_range(
+                ExpressionRef::Compare(left_compare),
+                expr_bool_op.as_any_node_ref(),
+                comment_ranges,
+                locator.contents(),
+            )
+            .is_some();
+            let right_has_paren = parenthesized_range(
+                ExpressionRef::Compare(right_compare),
+                expr_bool_op.as_any_node_ref(),
+                comment_ranges,
+                locator.contents(),
+            )
+            .is_some();
+
+            // Do not offer a fix if there are unbalanced parentheses
+            let fix = if left_has_paren == right_has_paren {
```

Comment:
Does this work in the case of (((a < b))) and (b < c)?

Reply:
No, which brings me back to not wanting to offer a fix here — I just want to fix the syntax bug and find a way to introduce a correct fix again separately.